applications/workspaces/backend/api/models/population.py

```diff
@@ -42,13 +56,35 @@ def save(
             self, force_insert=False, force_update=False, using=None, update_fields=None
     ):
         self.update_color()
+        has_file_changed = self._has_file_changed()
         super(Population, self).save(force_insert, force_update, using, update_fields)
-        split_cells_per_segment(self)
-        # TODO: also needs to generate new images
+        if has_file_changed:  # Do not recall the workflow on status changes
+            execute_generate_population_images_workflow(self.id)
+            # TODO: also add the following to argo workflows
```

Comment:
I might have gotten addicted to argo, but I think we could also split the csv's in a parallel task. You let me know @zsinnema

Reply:
@afonsobspinto seems like a plan. However I would suggest to use the pipeline operation to not consume too much of the cluster resources. When running in parallel every task will consume mem and cpu at the "same time". Since this is a async / background job it doesn't matter if it takes 20 seconds more....